py-polars/polars/eager/frame.py

```diff
@@ -1921,7 +1919,7 @@ def upsample(
         low = bounds["low"].dt[0]
         high = bounds["high"].dt[0]
         upsampled = pl.date_range(low, high, interval, name=by)
-        return pl.DataFrame(upsampled).join(self, on=by, how="left")
+        return pl.DataFrame(upsampled).join(self, on=by, how="left")  # type: ignore
```

Comment:
Thanks for that @ritchie46 . I wasn't too sure what was the preferred way to deal with that union...

Reply:
Yeah.. I am not entirely sure how to make mypy happy. But we clearly return a DataFrame. :)

Reply:
yeah, I think the best way would be to use [overload](https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading) on join. but its quite wordy...